read_files.py

```diff
@@ -0,0 +1,15 @@
+import os
+import mmap
+
+a=1
+for xxx in os.listdir("/home/ian/probanding/"):
+    b="No"
+    if "Player Name:" in xxx.read():
```

Comment:
Crei que aca esta la clave de todo el asunto.
Se pueden leer archivos de diferentes formas para diferentes propositos.
por ejemplo podes leer una pelicula (sea mpg o algo asi la extension) que es como un choclo
de bytes en GB, o podes leer un archivo que es la forma mas simple de texto (no importa la extension) de una linea.
Dependiendo de cosas asi como tamanio y codificacion vas a tener que elegir una u otra forma.
Esa forma en que lo estas haciendo es rara, te puedo expicar la forma en la que yo lo haria.

La cosa que iteras en listdir probablemente te devuelva un string, que probablemente seria el string que le corresponde al path (camino) por ejemplo: "/home/ian/Proyectos/play/elreadme.txt"
Nunca vi string.read() asi que poray te este devolviendo algo raro.
yo primero abriria el archivo asi:
```
archivo_abierto = open(path, 'r'): # esto es solo r y no rb, osea que lee de texto no de bytes
# do something with the opened file
archivo_abierto.close()
```

una archivo abierto de texto como lo de arriba se puede iterar, y devuelve lineas:

```
for linea in archivo_abierto:
    print(linea)
```

Reply:
si, se me complico cuando quise que el programa recorra varios archivos,

podria hacer un loop:

a = 1
loop:
    archivo_abierto = open("ejemplo"+str(a)+".txt", 'r')  # si cada archivo se llamara ejemplo1.txt / ejemplo2.txt / ejemplo3.txt
    # do something with the opened file
    archivo_abierto.close()
    a += 1

Reply:
y la forma mas pituca seria:

```
with open('/path/to/file', 'r') as the_file_opened:
    for linea the_file_opened:
        print(linea)
```

Reply:
pero ese loop recorre las lineas de un archivo, yo hablo de un loop que recorra varios archivos,

ej: archivo1.txt , archivo2.txt .... archivo 20.txt

Reply:
No me acuerdo bien, bien que era los que tenías que hacer, necesitas él loop por archivos, su necesitas leer varias líneas, necesitas otro loop para las líneas.

Reply:
lo que tengo que hacer:

un programa que recorra todos los archivos de un directorio, le cambie el nombre a cada uno (ejemplo: "ejemplo1.txt / ejemplo2.txt, etc) y a su vez que recorra el archivo y si encuentra un string "ejemplo:" le agregue a continuacion otro string.

cosas que ya aprendi:

como abrir 1 arhivo (pero no como recorrer varios)
como renombrar 1 archivo
como recorrer una linea (pero no buscar un string en esa linea, ni como agregar contenido en una posicion especifica)

Reply:
lo de listdir esta bien,
al iterarlo te deberia dar el path de cadaa archivo (probalo)
y con el path the cada archivo los podes abrir

Reply:
Quick update del codigo, funcionando para:
    - recorrer un directorio
    - renombrar cada archivo
    - agregar una linea al final del archivo

falta:
    - buscar un string
    - en caso de encontrarlo, agregar un string seguido al string encontrado


codigo:

import os

lista_archivos = os.listdir("/home/ian/Documentos/proyectos/textfiles/files/")
a=1
directorio = '/home/ian/Documentos/proyectos/textfiles/files/'

for archivo in lista_archivos:
    os.chdir(directorio)
    archivo_abierto = open(archivo,"a")
    archivo_abierto.write("Esta linea es agregada automaticamente, estas viendo el archivo numero " + str(a))
    archivo_abierto.close()
    nuevo_nombre = 'newname' + str(a) + '.txt'
    os.rename(directorio + archivo, directorio + nuevo_nombre)
    a += 1

Reply:
no se por que cuando lo copio aca, no se copia la identacion.